code/centauri/hyperspace/bin/chain.rs

```diff
@@ -562,6 +589,22 @@ impl Chain for AnyChain {
 			_ => unreachable!(),
 		}
 	}
+
+	async fn query_client_message(
+		&self,
+		host_block_hash: [u8; 32],
+		transaction_index: usize,
+		event_index: usize,
```

Comment:
this is still not chain agnostic, this should instead only take the block hash and consensus height.

Reply:
How about making it generic, so the client can use chain-specific types? Because otherwise, we'd have to run through all transactions in the block and then through all the messages in the `deliver` arguments